torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1208,6 +1208,17 @@ void ProcessGroupNCCL::waitForPendingWorks() {
 void ProcessGroupNCCL::enableCollectivesTiming() {
   enableTiming_.store(true);
 }
+
+bool ProcessGroupNCCL::tryWriteDebugInfo() {
```

Comment:
the naming of this suggests it will actually "try to write".  but what it actually does is just try to acquire a write lock.

it might be best to just wrap the creation of the thread inside this API, then the current name makes sense.  you could return the Thread object in every case, and if you do not get the lock you can just return an uninitialized Thread which the caller can determine is not joinable and it will know that's the equivalent of 'False' return value